src/index.js

```diff
@@ -207,10 +208,26 @@ function sanitizeHtml(html, options, _recursing) {
               }
             }
             if (name === 'iframe' && a === 'src') {
-              var whitelistedUrls = options.allowedIframeDomains.filter(function(url) {
-                return value.includes(url);
-              })
-              if (!whitelistedUrls.length) {
+              try {
+                var parsed = url.parse(value)
+                var whitelistedDomains = []
+                each(parsed, function(val, prop) {
+                  if (prop === 'hostname') {
+                    whitelistedDomains = options.allowedIframeDomains.filter(function(domain) {
+                      return val.includes(domain)
```

Comment:
You are still using `includes` so a name like "totallyfakeyoutube.com" would be accepted. You should be using a regular expression or otherwise matching either an exact match, or a match for a period followed by the domain name, at the end of the string only.

Reply:
I opted to go for an exact match to the hostname. I think the user needs to pass explicitly which hostname they would like to have matched. 

But, in some instances (like we see at our organization) a composer or redactor will stuff a localized url into the src so it looks something like `//www.youtube.com/embed/RTrYY5ycajY`, with no `http`. So the url parser doesn't pick up the host name, it just stuffs the whole url into the href property and makes host and hostname `null`. So I added some additional functionality to pull that apart using `split` and checking for an exact match again.